tests/contrib/kubernetes/kubernetes_request_factory/test_kubernetes_request_factory.py

```diff
@@ -0,0 +1,384 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.contrib.kubernetes.kubernetes_request_factory.\
+    kubernetes_request_factory import KubernetesRequestFactory
+from airflow.contrib.kubernetes.pod import Pod, Resources
+from airflow.contrib.kubernetes.secret import Secret
+import unittest
+
+
+class TestKubernetesRequestFactory(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.kubernetes_request_factory = KubernetesRequestFactory()
+
+        self.req = {
+            'apiVersion': 'v1',
+            'kind': 'Pod',
+            'metadata': {
+                'name': 'name'
+            },
+            'spec': {
+                'restartPolicy': 'Never',
+                'containers': [{
+                    'name': 'base',
+                    'image': 'airflow-worker:latest',
+                    'command': [
+                        "/usr/local/airflow/entrypoint.sh",
+                        "/bin/bash sleep 25"
+                    ],
+                }]
+            }
+        }
+
+    def test_extract_image(self):
+        input_req = self.req.copy()
+        reference = self.req.copy()
```

Comment:
```suggestion
        expected = self.req
```

We probably don't need to copy both of them - as it gets re-set by setUp for each test anyway.

Reply:
Good catch.